app/controllers/cases_controller.rb

```diff
@@ -554,40 +695,19 @@ def flag_for_clearance
     end
   end
 
-  def approve_response_interstitial
-    authorize @case, :can_approve_or_escalate_case?
-  end
-
-  def execute_response_approval
-    authorize @case
-
-    bypass_params_manager = BypassParamsManager.new(params)
-    service = CaseApprovalService.new(user: current_user, kase: @case, bypass_params: bypass_params_manager)
-    service.call
-    if service.result == :ok
-      current_team = CurrentTeamAndUserService.new(@case).team
-      if @case.ico?
-        flash[:notice] = t('notices.case/ico.case_cleared')
-      else
-        flash[:notice] = t('notices.case_cleared', team: current_team.name,
-                                                   status: I18n.t("state.#{@case.current_state}").downcase)
-      end
-      redirect_to case_path(@case)
-    else
-      flash[:alert] = service.error_message
-      render :approve_response_interstitial
-    end
-  end
-
   def request_amends
     authorize @case, :execute_request_amends?
     @next_step_info = NextStepInfo.new(@case, 'request-amends', current_user)
   end
 
   def execute_request_amends
     authorize @case
-    CaseRequestAmendsService.new(user: current_user, kase: @case, message: params[:case][:request_amends_comment]).call
-    if @case.type_abbreviation == 'SAR'
+    CaseRequestAmendsService.new(
+        user: current_user,
+        kase: @case,
+        message: params[:case][:request_amends_comment],
+        is_compliant: params[:case][:draft_compliant] == 'yes').call
+    if @case.sar?
       flash[:notice] = 'Information Officer has been notified a redraft is needed.'
```

Comment:
Old code - but would prefer these flash strings to be moved into the translation file considering the rest of the function has changed anyway.

Reply:
OOS for this ticket